### PR TITLE
A bash snippet to get the directory a script is located in

### PR DESCRIPTION
**Pull request by omajid**
See https://stackoverflow.com/a/246128/3561275 for details about how this works.

I end up needing this almost every other day or so. For it's usefulness, please note that it's the [highest voted bash question on StackOverflow, ever](https://stackoverflow.com/questions/tagged/bash?sort=votes). And probably the only question in the top 10 or so where the answer is a recipe.

### CODE REVIEW

**Comment by AndreaCrotti** [comment]
This one is a bit strange, since it's not really a template, but just a snippet of code.
I suppose given how generally useful it seems to be that we can just make it a snippet, in case someone needs it.

File: snippets/sh-mode/script-dir [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# -*- mode: snippet -*-
+# name: the currently executing/sourced script's directory
+# key: script-dir
+# --
+SOURCE="\${BASH_SOURCE[0]}"
```
Comment (AndreaCrotti) [inline snippets/sh-mode/script-dir]: could you link the stackoverflow question where you got this from (even as a comment which will end up in the template is ok I believe)?

**Comment by omajid** [comment]
> This one is a bit strange, since it's not really a template, but just a snippet of code.

I based this on https://github.com/AndreaCrotti/yasnippet-snippets/blob/master/snippets/sh-mode/safe-bang which is very similar in practice.

File: snippets/sh-mode/script-dir (hunk 1 above)
Comment (omajid) [reply snippets/sh-mode/script-dir]: Done.

**Comment by AndreaCrotti** [comment]
Sure why not it can be useful, thanks

**Pull request merged**